pinot-tools/src/main/java/com/linkedin/pinot/tools/admin/command/MoveReplicaGroup.java

```diff
@@ -0,0 +1,434 @@
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *         http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.linkedin.pinot.tools.admin.command;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.google.common.base.Function;
+import com.google.common.base.Joiner;
+import com.google.common.base.Preconditions;
+import com.linkedin.pinot.common.config.AbstractTableConfig;
+import com.linkedin.pinot.common.utils.CommonConstants;
+import com.linkedin.pinot.common.utils.helix.HelixHelper;
+import com.linkedin.pinot.common.utils.retry.RetryPolicies;
+import com.linkedin.pinot.tools.Command;
+import com.linkedin.pinot.tools.PinotZKChanger;
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.Charset;
+import java.nio.file.Files;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.Arrays;
+import java.util.Comparator;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.PriorityQueue;
+import javax.annotation.Nullable;
+import org.apache.helix.PropertyPathConfig;
+import org.apache.helix.PropertyType;
+import org.apache.helix.ZNRecord;
+import org.apache.helix.manager.zk.ZKHelixAdmin;
+import org.apache.helix.manager.zk.ZNRecordSerializer;
+import org.apache.helix.model.IdealState;
+import org.apache.helix.store.zk.ZkHelixPropertyStore;
+import org.json.JSONException;
+import org.kohsuke.args4j.CmdLineException;
+import org.kohsuke.args4j.CmdLineParser;
+import org.kohsuke.args4j.Option;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+
+/**
+ * Command to migrate a subset of replica group from current servers to the provided destination servers.
+ * This command is intended to be run multiple times to migrate all the replicas of a table to the destination
+ * servers (if intended).
+ */
+public class MoveReplicaGroup extends AbstractBaseAdminCommand implements Command {
+  private static Logger LOGGER = LoggerFactory.getLogger(MoveReplicaGroup.class);
+
+  @Option(name = "-srcHostsFile", aliases = {"-s", "--src"}, required = true, metaVar = "<filePath>", usage = "File with names of source hosts")
+  private String srcHostsFile;
+
+  @Option(name = "-destHostsFile", aliases = {"-d", "--dest"}, required = false, metaVar = "<File Path>",
+      usage = "File with destination servers list")
+  private String destHostsFile = "";
+
+  @Option(name = "-tableName", aliases = {"-t", "-table"}, required = true, metaVar = "<string>", usage = "Table name")
```

Comment:
Added explanation. I wanted to tool to just work so validateParams does the checking right now. 

It does not support realtime because I haven't thought through all cases yet but we can extend to support it